padle/padle.py

```diff
@@ -0,0 +1,801 @@
+            f"Here are your guesses:\n{json.dumps(all_guesses)}"
+        else:
+            data = f"No data is stored for user with ID {user_id}."
+        return {"user_data.txt": BytesIO(data.encode())}
+
+    async def red_delete_data_for_user(self, *, requester, user_id):
+        """Delete a user's personal data."""
+        await self.config.user_from_id(user_id).clear()
+
+    async def register_menu(self):
+        await self.bot.wait_until_ready()
+        menulistener: Any = self.bot.get_cog("MenuListener")
+        if menulistener is None:
+            logger.warning("MenuListener is not loaded.")
+            return
+        await menulistener.register(self)
+
+    async def get_dbcog(self) -> "DBCog":
+        dbcog: Any = self.bot.get_cog("DBCog")
+        if dbcog is None:
+            raise ValueError("DBCog cog is not loaded")
+        await dbcog.wait_until_ready()
+        return dbcog
+
+    def cog_unload(self):
+        self._daily_padle_loop.cancel()
+
+    async def get_menu_default_data(self, ims):
+        user = self.bot.get_user(ims['original_author_id'])
+        data = {
+            'dbcog': await self.get_dbcog(),
+            'user_config': await BotConfig.get_user(self.config, ims['original_author_id']),
+            'today_guesses': await self.get_today_guesses(user, ims.get('current_day'))
+        }
+        return data
+
+    async def get_today_guesses(self, user: discord.User, current_day: Union[str, int]):
+        if current_day is None:
+            return None
+        if user is None:
+            return {}
+        async with self.config.user(user).all_guesses() as all_guesses:
+            return all_guesses.get(str(current_day))
+
+    @commands.group()
+    async def padle(self, ctx):
+        """Commands pertaining to PADle"""
+
+    @commands.group(aliases=['padleconfig'])
+    @auth_check('padleadmin')
+    async def padleadmin(self, ctx):
+        """Commands pertaining to PADle setup"""
+
+    @padle.command()
+    async def help(self, ctx):
+        """Instructions for PADle"""
+        prefix = ctx.prefix
+        await ctx.send("- PADle is similar to Wordle, except for PAD cards.\n"
+                       "- You have infinite tries to guess the hidden PAD card (chosen from a list "
+                       "of more well-known monsters). Everyone is trying to guess the same PAD card. "
+                       "With each guess, you are given feedback as to how similar the two cards are, "
+                       "including comparing the awakenings, rarity, typings, attributes, and monster "
+                       "point sell value.\n- A new PADle is available every day. "
+                       "Use `{}padle start` to begin!".format(prefix))
+
+    @padle.command(aliases=["sub"])
+    async def subscribe(self, ctx, sub_arg: bool = True):
+        """Subscribe to daily notifications of new PADles"""
+        prefix = ctx.prefix
+        subbed_users = await self.config.subs()
```

Comment:
If you're using subbed_users and editing it, you can put the `async with` suite up here so you only have to read from the file once.